front/src/actions/message.js

```diff
@@ -43,13 +43,18 @@ function createActions(store) {
       });
     },
     pushMessage(state, message) {
+      let newMessages = store.getState().messages;
+      // Check if message is already in the list
+      if (message.id && newMessages.filter(m => m.id === message.id).length) {
```

Comment:
.filter will scan the whole array of message, it could be long just to verify that the message is absent! 

I recommend using .find() instead (https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Objets_globaux/Array/find). This function is optimized and will stop when the result was found

Reply:
Modified to use find()